### PR TITLE
ignore result of function to compile on G++ 5

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/2785

### CODE REVIEW

File: sdk/core/azure-core/src/base64.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ namespace Azure { namespace Core {
     bio = BIO_push(BIO_new(BIO_f_base64()), bio);
     BIO_set_flags(bio, BIO_FLAGS_BASE64_NO_NL);
     BIO_write(bio, data.data(), static_cast<int>(data.size()));
-    BIO_flush(bio);
+    (void)BIO_flush(bio);
```
Comment (RickWinter) [inline sdk/core/azure-core/src/base64.cpp:67]: This return shouldn't be removed.  We need to check the return codes and act accordingly.  I've opened a new issue to remove this dependency all out.
https://github.com/Azure/azure-sdk-for-cpp/issues/2790

**Review by RickWinter — APPROVED** [review]

File: sdk/core/azure-core/src/base64.cpp (hunk 1 above)
Comment (vhvb1989) [reply sdk/core/azure-core/src/base64.cpp:67]: yeah, I kind of remember this was mentioned before. I think @ahsonkhan moved this from storage to core and would be eventually be replaced for the base64 implementation that @antkmsft  wrote for C99.  If that's still the plan, the change could be targeted for Sep release maybe.

**Comment by vhvb1989** [comment]
Public pipeline g++-5 : 
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1063962&view=logs&j=efb8e22a-57a5-5d01-2d97-95ae9b6a04dd&t=a090feb1-5f4d-5f84-82e3-a9887e35fb99&l=15

Internal pipelines also building with g++ with tests

**Pull request merged**